sys/arduino/serialport.cpp

```diff
@@ -41,19 +42,21 @@ SerialPort::SerialPort(uart_t dev)
 
 int SerialPort::available(void)
 {
-    return (int)rx_buf.avail;
+    return stdio_avail();
 }
 
 void SerialPort::begin(long baudrate)
 {
+    stdio_init();
     /* this clears the contents of the ringbuffer... */
-    ringbuffer_init(&rx_buf, rx_mem, SERIAL_RX_BUFSIZE);
-    uart_init(dev, (uint32_t)baudrate, rx_cb, (void *)&rx_buf);
+    stdio_clear();
```

Comment:
In case of `stdio_uart`, it might be enough to call `stdio_init`. It depends on `periph_uart` implementation.